platform/iphone/export/export.cpp

```diff
@@ -207,7 +207,9 @@ Error EditorExportPlatformIOS::export_project(const Ref<EditorExportPreset> &p_p
 	int total_size = 0;
 
 	while (ret == UNZ_OK) {
+#ifdef OSX_ENABLED
```

Comment:
More than adding guards everywhere this bool is defined, would be good to just make it:
```
if (is_execute) {
#ifdef OSX_ENABLED // or better, UNIX_ENABLED?
				// we need execute rights on this file
				chmod(file.utf8().get_data(), 0755);
#else
				// print some info message that the binary can't be made executable so it should be done manually on a UNIX system
#endif
}
```